code/modules/projectiles/guns/projectile/rechargable_magazine.dm

```diff
@@ -65,11 +65,22 @@
 	w_class = 5
 	var/obj/item/weapon/gun/projectile/minigun/gun = null
 	var/armed = 0 //whether the gun is attached, 0 is attached, 1 is the gun is wielded.
+	var/overheat = 0
+	var/overheat_max = 40
+	var/heat_diffusion = 1
 
 /obj/item/weapon/minigunpack/New()
 	gun = new(src)
+	SSobj.processing += src
```

Comment:
fast process this bad boy and crank up the values, and put a bar sprite on the minigun showing the overheat status